### DISCUSSION TITLE
Enhance the way that relationship field values are fetched in UI

### DISCUSSION THREAD

**Discussion by jmikrut** [payloadcms/payload#1831] [Roadmap]
Right now, each relationship field is responsible for fetching its own related data, but this can be problematic and cause performance issues if you have many relationship fields with hundreds of related documents. 

There is also a potential problem with URL length limits if you do have a relationship field that fetches 300+ relations.

There may be a way to create a top-level relationship context "store" which stores related documents required in a view by ID, and could be checked to see if a document exists before fetching its data.

There's a good amount of complexity in this, but it could offer some performance enhancements if we address it correctly.

**Comment by wrobson-lllow** [comment]
Hi  @jmikrut ,
I came across this while trying to find a solution to me recent issue [found here](https://github.com/payloadcms/payload/discussions/2825) .
If I search for a particular record that has not already been fetched and appearing in the dropdown, I get "no options" and then the dropdown won't fetch anymore records. The only way to find the appropriate record is to scroll down the list. 
Do you have any idea what could be causing this? Or do you perhaps have a potential workaround?
